libbeat/common/file/rotator.go

```diff
@@ -202,25 +218,134 @@ func (r *Rotator) Write(data []byte) (int, error) {
 		if err := r.openNew(); err != nil {
 			return 0, err
 		}
-	} else if r.intervalRotator != nil && r.intervalRotator.NewInterval() {
-		if err := r.rotate(rotateReasonTimeInterval); err != nil {
-			return 0, err
+	} else {
+		if reason, t := r.isRotationTriggered(dataLen); reason != rotateReasonNoRotate {
+			if err := r.closeFile(); err != nil {
```

Comment:
we do close+rot.Rotate but we have r.Rotate Func which does the same

Reply:
`r.Rotate` is used to trigger manual rotations. But I can refactor the code a bit more so `r.Rotate` and this part uses the same function.